specs/TransportProtocols.md

```diff
@@ -0,0 +1,175 @@
+# Transport Protocols
+
+This document describes the protocols used by the three ASP.NET Endpoint Transports: WebSockets, Server-Sent Events, Long Polling and HTTP Post
+
+## Transport Requirements
+
+A transport is required to have the following attributes:
+
+1. Duplex - Able to send messages from Server to Client and from Client to Server
+1. Frame-based - Messages have fixed length (as opposed to streaming, where data is just pushed throughout the connection)
+1. Frame Metadata - Able to encode two pieces of Frame Metadata
+    * `Type` - Either `Binary` or `Text`.
+    * `EndOfMessage` - Boolean indicating if this frame represents the final frame of a complete message.
```

Comment:
How much do we care about "EndOfMessage" here? We want to represent this data over a transport like WebSockets which provides it, but SignalR certainly doesn't care about it. It seems reasonable to me that we could just ignore it in the other transports. But the issue does arise when someone puts a partial message in the Channel, and the Long-Polling or SSE transport is attached. If we don't represent this data, are we misrepresenting the user's intent?

Despite describing how to represent it here, I actually think I fall in the camp of only having WebSockets support this. If the other transports receive a "partial" message, they just render it and ignore the EndOfMessage flag. However, we definitely need to propagate this flag up from/to transports that do support it (i.e. WebSockets) in order to allow ASP.NET Endpoints to be able to serve any kind of WebSockets-based protocol.